### PR TITLE
HDDS-5385. [FSO] Remove ozone.om.metadata.layout config in OM

### PR DESCRIPTION
**Pull request by JyotinderSingh**
## What changes were proposed in this pull request?

This task is to remove the metadata layout config as bucket type will be an argument. Also, will define `ozone.bucket.type.default` to handle a case where the user is not passing any args.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5385

## How was this patch tested?

related integration tests

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/nssummary/NSSummaryAdmin.java [hunk 1]
```diff
@@ -74,6 +88,30 @@ public Void call() throws Exception {
     return OzoneAdmin.class;
   }
 
+  public boolean isFSOEnabled(String path) throws IOException {
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/nssummary/NSSummaryAdmin.java:87]: Please rename function name to` isFileSystemOptimizedBucket(String path)`

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestReadRetries.java [hunk 2]
```diff
@@ -116,7 +118,7 @@ public void init() throws Exception {
     OzoneConfiguration conf = new OzoneConfiguration();
     conf.setInt(ScmConfigKeys.OZONE_SCM_PIPELINE_OWNER_CONTAINER_COUNT, 1);
     TestOMRequestUtils.configureFSOptimizedPaths(conf,
-            true, layoutVersion);
+            true, BucketLayout.fromString(layoutVersion));
```
Comment (rakeshadr) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestReadRetries.java:121]: Please rename var as well. `layoutVersion` ->to-> `bucketLayout`

**Review by rakeshadr — COMMENTED** [review]
Thanks @JyotinderSingh for the good work. Added few comments, please go through it.

**Comment by JyotinderSingh** [comment]
Thank you for the reviews @rakeshadr. I have addressed your comments in my latest commit.

**Review by rakeshadr — APPROVED** [review]
+1 LGTM, thanks @JyotinderSingh 

Pending build CI

**Comment by rakeshadr** [comment]
Thanks @JyotinderSingh for rebased the patch. We got a clean build and I'm merging the PR.

**Pull request merged**